simpleclient/src/main/java/io/prometheus/client/Summary.java

```diff
@@ -138,22 +138,44 @@ protected Child newChild() {
     return new Child(quantiles, maxAgeSeconds, ageBuckets);
   }
 
+
+  public static class SimpleTimer {
+    private final long start;
+    private final TimeProvider timeProvider;
+
+    public SimpleTimer(TimeProvider timeProvider) {
+      this.timeProvider = timeProvider;
+      start = timeProvider.nanoTime();
+    }
+
+    public SimpleTimer() {
+      this(Child.timeProvider);
+    }
+
+    /**
+     * @return Measured duration in seconds since {@link SimpleTimer} was constructed.
+     */
+    public double elapsedSeconds() {
+      return (timeProvider.nanoTime() - start) / NANOSECONDS_PER_SECOND;
+    }
+  }
+
   /**
    * Represents an event being timed.
    */
   public static class Timer {
     private final Child child;
-    private final long start;
+    private final SimpleTimer innerTimer;
     private Timer(Child child) {
       this.child = child;
-      start = Child.timeProvider.nanoTime();
+      innerTimer = new SimpleTimer(Child.timeProvider);
```

Comment:
An additional new object on each event (of which there are likely many for a single request) seems like it might get expensive.